src/io_callbacks.zig

```diff
@@ -0,0 +1,463 @@
+const std = @import("std");
+const assert = std.debug.assert;
+const os = std.os;
+const linux = os.linux;
+const IO_Uring = linux.IO_Uring;
+const io_uring_cqe = linux.io_uring_cqe;
+const io_uring_sqe = linux.io_uring_sqe;
+
+pub const IO = struct {
+    /// An intrusive first in/first out linked list.
+    const Fifo = struct {
```

Comment:
Let's do `FIFO` here instead of `Fifo` in keeping with `IO` vs `Io`.

Reply:
Happy to change that for consistency, I've been trying to follow the style of the project so far even though it conflicts with the official zig style guide on this point and on the naming of functions (the zig style guide and standard library use camelCase instead of snake_case).

As for the precise algorithm used to name types this is what has been settled on (though it could almost certainly be better documented and is not yet followed everywhere in the standard library) https://github.com/ziglang/zig/issues/1097#issuecomment-722112034.

I personally try to waste as little time as possible on coding style, all I want is consistency, so I'm happy to go with whatever you prefer here.

Reply:
Great, yes I was following that issue too, and had hoped all "the ayes would have it" in favor of snake_case functions but alas! It's also a pretty big change for Zig to make on behalf of everyone if it had gone through. I noticed that you also stuck to all the current styles of the project, thanks.

I'm happy that we keep the snake_case method names within the project, because that extra separator between words helps readability, especially in our VR implementation so far where we have methods like `find_latest_header_break_between`... it's the closest one can get to spaces but without using them. That was also something I always liked about C and Ruby.

> I personally try to waste as little time as possible on coding style, all I want is consistency, so I'm happy to go with whatever you prefer here.

Yes, same here. That's also why I find camelCase difficult, nobody knows whether it's XmlHttpRequest or XMLHTTPRequest etc. as that issue highlighted.

But in this case I actually really just thought FIFO here because it's an abbreviation! 😄

Reply:
> Yes, same here. That's also why I find camelCase difficult, nobody knows whether it's XmlHttpRequest or XMLHTTPRequest etc. as that issue highlighted.

The algorithm described in the comment I linked resolves this ambiguity, but point taken. I'll rename to FIFO.